### ISSUE TITLE
Abotu BEVFormer-S

### ISSUE THREAD

**Issue by LuckySunnye** [zhiqi-li/BEVFormer#53]
Thanks for your great job!
I want to training BEVFormer without temporal information, can you tell me how to adjust config.py and code to reproduce the results of BEVFormer-S in your paper?
![image](https://user-images.githubusercontent.com/96170598/176980006-2a242b17-9c65-49c6-a139-bfc938f56e60.png)

**Comment by Bosszhe** [comment]
I have the same question and hope to give a tutorial about training BEVFormer-S, thx.

**Comment by yeeoon** [comment]
I also have the same question. Can I get a tutorial to reproduce the BEVFormer-S?

**Comment by czy341181** [comment]
Hi! thanks for your great work. 
I also want to reproduce the result of BEVFormer-s. And I get the following result, which is lower than paper. 

                               NDS          map
origin paper          0.4480        0.375
BEVFormer-S.      0.3920         0.3059

**Comment by zhiqi-li** [comment]
> I modify the detectors/bevformer.py forward_train function :
For bevformer-S, you should set `video_test_mode=False` in your config.

**Comment by Roserland** [comment]
> 

So, the result of original paper is trained under `TemporalSelfAttention` while tested under degraded SelfAttention(with `video_test_mode=False` in config)

**Comment by zhiqi-li** [comment]
> > 
> 
> So, the result of BEVFormer-S in original paper is trained under `TemporalSelfAttention` while tested under degraded SelfAttention(with `video_test_mode=False` in config) ?

NO, BEVFormer-S was trained with only one sample without temporal information, and in this case, `TemporalSelfAttention` is just one self-attention.  So that, for BEVFormer-S,  you should set  `video_test_mode=False` while testing.

**Comment by czy341181** [comment]
Thanks for your timely reply. I got the results of paper.

**Comment by Roserland** [comment]
> > > 
> > 
> > 
> > So, the result of BEVFormer-S in original paper is trained under `TemporalSelfAttention` while tested under degraded SelfAttention(with `video_test_mode=False` in config) ?
> 
> NO, BEVFormer-S was trained with only one sample without temporal information, and in this case, `TemporalSelfAttention` is just one self-attention. So that, for BEVFormer-S, you should set `video_test_mode=False` while testing.

Thank you! And Excellent work!

**Comment by JudasDie** [comment]
> Thanks for your timely reply. I got the results of paper.

Hi, could you kindly explain how to get the results of the paper, by training with `video_test_mode=False`, `len_queue=1` and testing with `video_test_mode=False`? Or other setting?

**Comment by Daniel-xsy** [comment]
@czy341181 Do you train BEVFormer-S from scratch and get the results of the paper?

**Comment by wenyuqing** [comment]
> > > 
> > 
> > 
> > So, the result of BEVFormer-S in original paper is trained under `TemporalSelfAttention` while tested under degraded SelfAttention(with `video_test_mode=False` in config) ?
> 
> NO, BEVFormer-S was trained with only one sample without temporal information, and in this case, `TemporalSelfAttention` is just one self-attention. So that, for BEVFormer-S, you should set `video_test_mode=False` while testing.

Hi， do you mean set the data queue length to 1 to train static version while keep all other settings the same with the temporal version？
